src/CodeGeneration.Roslyn/TransformationContext.cs

```diff
@@ -13,20 +11,28 @@ public class TransformationContext
         /// <param name="processingMember">The syntax node the generator attribute is found on.</param>
         /// <param name="semanticModel">The semantic model.</param>
         /// <param name="compilation">The overall compilation being generated for.</param>
-        public TransformationContext(MemberDeclarationSyntax processingMember, SemanticModel semanticModel, CSharpCompilation compilation)
+        /// <param name="projectDirectory">The absolute path of the directory where the project file is located</param>
+        public TransformationContext(CSharpSyntaxNode processingMember, SemanticModel semanticModel, CSharpCompilation compilation,
+                                     string projectDirectory)
         {
             ProcessingMember = processingMember;
             SemanticModel = semanticModel;
             Compilation = compilation;
+            ProjectDirectory = projectDirectory;
         }
 
         /// <summary>Gets the syntax node the generator attribute is found on.</summary>
-        public MemberDeclarationSyntax ProcessingMember { get; }
+        public CSharpSyntaxNode ProcessingMember { get; }
```

Comment:
I don't think all nodes are "members", so this property could use a new name. And to avoid backward breaking changes, can you create a new member rather than change the existing one?

Reply:
All `MemberDeclarationSyntax` are `CSharpSyntaxNode`. Because in all practical examples they cast it to exact class it's backward-compatible, I guess. And I don't believe we should have to same nodes (referencing the same object) but where one property is of more specific type.

Reply:
Thanks. That's better.
Regarding the breaking change I mentioned, changing the type of a public member is a binary breaking change. But I guess for this project in particular, that's less interesting than a source breaking change. And we've made a few of those in the past, so I guess we're good.